//
// Copyrights (c) 2015, Juniper Networks, Inc.
// All rights reserved.
//

//  
// Licensed to the Apache Software Foundation (ASF) under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  The ASF licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.
//

//
// Nitin Kumar, 2015
//
// This file defines generic protofile used by JUNOS Routing Engine
// based sensors
//
// Version 1.0
//

syntax = "proto2";
import "telemetry_top.proto";

message keyvalue {
    optional string key = 1;
    optional int32 int32_val = 2;
    optional uint64 uint64_val = 3;
    optional string string_val = 4;
}

message kvresponse {
    repeated keyvalue kvpairs = 1;
}

//
// this occupies branch 2000 from JuniperNetworksSensors
//
extend JuniperNetworksSensors {
    optional kvresponse kvresponseExt = 2000;
}

